src/components/portal/portal-arrow.scss

```diff
@@ -0,0 +1,94 @@
+/**
+ * @prop --portal-arrow-color: The color of the arrow has to be the same as the surface opened in the portal.
+ */
+
+$size-of-arrow: 0.5rem;
+
+limel-portal-arrow,
+limel-portal-arrow::before {
+    position: absolute;
+}
+
+limel-portal-arrow {
+    visibility: hidden;
+
+    &::before {
+        visibility: visible;
+        content: '';
+
+        border: {
+            width: $size-of-arrow;
+            color: transparent;
+            style: solid;
+        }
+        [data-popper-placement^='top'] & {
+            border-top-color: var(--portal-arrow-color);
+            margin-left: -$size-of-arrow;
+        }
+        [data-popper-placement^='bottom'] & {
+            border-bottom-color: var(--portal-arrow-color);
+            margin-left: -$size-of-arrow;
+        }
+        [data-popper-placement^='left'] & {
+            border-left-color: var(--portal-arrow-color);
+            margin-top: -$size-of-arrow;
+        }
+        [data-popper-placement^='right'] & {
+            border-right-color: var(--portal-arrow-color);
+            margin-top: -$size-of-arrow;
+        }
+    }
+
+    [data-popper-placement^='top'] & {
+        bottom: 0;
+    }
+    [data-popper-placement^='bottom'] & {
+        top: -$size-of-arrow * 2;
+    }
+    [data-popper-placement^='left'] & {
+        right: 0;
+    }
+    [data-popper-placement^='right'] & {
+        left: -$size-of-arrow * 2;
+    }
+}
+
+.limel-portal--container {
+    &.has-arrow {
+        &[data-popper-placement^='right'] {
+            margin-left: $size-of-arrow !important;
+        }
+        &[data-popper-placement^='left'] {
+            margin-right: $size-of-arrow !important;
+        }
+        &[data-popper-placement^='top'] {
+            margin-bottom: $size-of-arrow !important;
+        }
+        &[data-popper-placement^='bottom'] {
+            margin-top: $size-of-arrow !important;
+        }
+    }
+}
+
+.arrow-style-tooltip::before {
```

Comment:
Same here, portal should know that tooltip is a consumer. We should inject these styles somehow instead.

Reply:
… while the z-index is easy for tooltip too, the fact that we use a keyframed animation is making this case pretty complicated for me. I added some properties to allow injecting the necessary styles, but then released that because neither the container nor the arrow are using shadow-dom, the styles "leak" into the rest of the document. So I set about trying to fix that, and I've managed to fix it so it's working for popover and tooltip, but now limel-menu is completely broken. It's not even possible to click on a menu item 😭

(If you ever wonder why I keep nagging about keeping complexity as low as possible, this is one of the reasons; in complex code, changes often have complicated side-effects 😞)

I don't really know where to go with this at the moment. The portal component is very messy already before this arrow addition, and the code is very hard to follow. This PR doesn't really make it _worse_, but it adds code to the mess, which just makes the mess _bigger_. I hate it when people have put in a lot of hard work and then I have to come and say that "no we can't merge this, because blah blah", none which is the fault of the people working on the thing I'm reviewing 😭, but still, I think this needs some more work before it should be merged, work I unfortunately don't have time to do before going on vacation again 😞

Since there isn't really anyone to hand this over to right now, and I'm off on vacation from tomorrow, I'm ok with having a call with someone on one of my vacation days, to explain and hand this over. Just send me a direct message in Slack!